week1/reposts/boccato/Main.hs

```diff
@@ -0,0 +1,38 @@
+module Main where
+
+import Control.Monad
+import Data.List
+import System.Random
+import Text.Printf
+
+nWinners = 7
+nSimulations = 10000
+nCurrentReposts = 1000000
+
+takeDistinct :: Eq a => Int -> [a] -> [a]
```

Comment:
A warning: `nub` is quadratic. In this case it doesn't matter, I think, because the list is short, but generally you should be aware of it.